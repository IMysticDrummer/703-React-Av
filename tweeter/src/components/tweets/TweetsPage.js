import { useEffect } from 'react';
//import { useState } from 'react';
import { Link } from 'react-router-dom';

//import { getLatestTweets } from './service';
import classNames from 'classnames';

import Tweet from './Tweet';
import styles from './TweetsPage.module.css';
import Page from '../layout/Page';
import Button from '../common/Button';
//Guardar los tweets en Redux
//import { tweetsLoaded } from '../../store/actions';
import { tweetsLoad } from '../../store/actions';
import { connect } from 'react-redux';
import { getTweets } from '../../store/selectors';

const EmptyList = () => (
  <div style={{ textAlign: 'center' }}>
    <p>Be the first one!</p>
    <Button
      as={Link}
      to='/tweets/new'
      variant='primary'>
      Create tweet
    </Button>
  </div>
);

//Guardar los tweets en Redux
// const useTweets = () => {
//   const [tweets, setTweets] = useState([]);

//   useEffect(() => {
//     const execute = async () => {
//       const tweets = await getLatestTweets();
//       setTweets(tweets);
//     };
//     execute();
//     // getLatestTweets().then(tweets => {
//     //   setTweets(tweets);
//     // });
//   }, []);

//   return tweets;
// };

//Cambios para usar Redux para guardar los tweets
// const TweetsPage = props => {
//   const tweets = useTweets();

export const TweetsPage = ({ onTweetsLoaded, tweets, ...props }) => {
  // useEffect(() => {
  //   const execute = async () => {
  //     const tweets = await getLatestTweets();
  //     onTweetsLoaded(tweets);
  //   };
  //   execute();
  // }, []);
  useEffect(() => {
    onTweetsLoaded();
  }, []);

  const className = classNames(styles.tweetsPage, {
    [styles.empty]: !tweets.length,
  });

  return (
    <Page
      title="What's going on..."
      {...props}>
      <div className={className}>
        {tweets.length ? (
          <ul>
            {tweets.map((tweet) => (
              <li key={tweet.id}>
                <Link to={`/tweets/${tweet.id}`}>
                  <Tweet {...tweet} />
                </Link>
              </li>
            ))}
          </ul>
        ) : (
          <EmptyList />
        )}
      </div>
    </Page>
  );
};

//Usar connect
//export default TweetsPage;

const mapStateToProps = (state, ownProps) => ({
  tweets: getTweets(state),
});

// const mapDispatchToProps = (dispatch, ownProps) => ({
//   onTweetsLoaded: tweets => dispatch(tweetsLoaded(tweets)),
// });

const mapDispatchToProps = {
  //onTweetsLoaded: tweetsLoaded,
  onTweetsLoaded: tweetsLoad,
};

const connectedTweetsPage = connect(
  mapStateToProps,
  mapDispatchToProps
)(TweetsPage);

export default connectedTweetsPage;
