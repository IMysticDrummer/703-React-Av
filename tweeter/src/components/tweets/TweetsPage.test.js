import { render } from '@testing-library/react';
import { TweetsPage } from './TweetsPage';
import { BrowserRouter as Router } from 'react-router-dom';

describe('TweetsPage', () => {
  const defaultProps = {
    onTweetsLoaded: jest.fn(),
    tweets: [],
  };
  const renderComponent = () =>
    render(
      <Router>
        <TweetsPage {...defaultProps} />{' '}
      </Router>
    );

  test('snapshot without tweets', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  test('snapshot with tweets', () => {
    const tweets = [
      {
        id: 1,
        content: 'hello',
        updateAt: '2022-12-20',
        likes: [],
        user: { name: 'david', username: 'david' },
      },
    ];
    const { container } = renderComponent({ tweets });
    expect(container).toMatchSnapshot();
  });

  test('should call onTweetsLoaded', () => {
    renderComponent();
    expect(defaultProps.onTweetsLoaded).toHaveBeenCalled();
  });
});
