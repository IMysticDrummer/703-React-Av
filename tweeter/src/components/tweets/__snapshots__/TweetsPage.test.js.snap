// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetsPage snapshot with tweets 1`] = `
<div>
  <h2
    class="layout-title bordered"
  >
    What's going on...
  </h2>
  <section
    class="layout-content"
  >
    <div
      class="tweetsPage empty"
    >
      <div
        style="text-align: center;"
      >
        <p>
          Be the first one!
        </p>
        <a
          class="sc-bcXHqe cZfeII"
          href="/tweets/new"
          variant="primary"
        >
          Create tweet
        </a>
      </div>
    </div>
  </section>
   
</div>
`;

exports[`TweetsPage snapshot without tweets 1`] = `
<div>
  <h2
    class="layout-title bordered"
  >
    What's going on...
  </h2>
  <section
    class="layout-content"
  >
    <div
      class="tweetsPage empty"
    >
      <div
        style="text-align: center;"
      >
        <p>
          Be the first one!
        </p>
        <a
          class="sc-bcXHqe cZfeII"
          href="/tweets/new"
          variant="primary"
        >
          Create tweet
        </a>
      </div>
    </div>
  </section>
   
</div>
`;
